lib/common.js

```diff
@@ -9,18 +9,30 @@ var debug = require('debug')('nock.common');
  * @param  {Object} options - a parsed options object of the request
  */
 var normalizeRequestOptions = function(options) {
-  options.proto = options.proto || (options._https_ ? 'https': 'http');
-  options.port = options.port || ((options.proto === 'http') ? 80 : 443);
-  if (options.host) {
-    debug('options.host:', options.host);
-    if (! options.hostname) {
-      if (options.host.split(':').length == 2) {
-        options.hostname = options.host.split(':')[0];
-      } else {
-        options.hostname = options.host;
+  if (options._key) {
+    // the most interceptor info is contained in _key
+    var parts = /(\w+) (\w+):\/\/([^:]+)\/?(.*)?:(\d+)\/?(.*)/.exec(options._key);
+    options.method = parts[1];
```

Comment:
I hate that we're currently changing the options in place. We should instead be creating and returning a new object..